src/main/java/uk/gov/justice/hmpps/prison/service/PrisonerReleaseAndTransferService.java

```diff
@@ -0,0 +1,238 @@
+
+        final var releaseDateTime = getAndCheckMovementTime(requestToReleasePrisoner.getReleaseTime(), booking.getBookingId());
+        // set previous active movements to false
+        final Long bookingId = setPreviousMovementsToInactive(booking);
+
+        createOutMovement(booking, REL, movementReason, toLocation, releaseDateTime, requestToReleasePrisoner.getCommentText(), null);
+
+        // generate the release case note
+        generateReleaseNote(booking, releaseDateTime, movementReason);
+
+        // Update occupancy (recursively)
+        updateBeds(booking, releaseDateTime);
+
+        deactivateSentences(bookingId);
+
+        // update the booking record
+        booking.setInOutStatus("OUT");
+        booking.setActiveFlag("N");
+        booking.setBookingStatus("C");
+        booking.setLivingUnitMv(null);
+        booking.setAssignedLivingUnit(null);
+        booking.setLocation(toLocation);
+        booking.setBookingEndDate(releaseDateTime);
+        booking.setStatusReason(REL.getCode() + "-" + movementReasonCode);
+        booking.setCommStatus(null);
+
+    }
+
+    public void transferOutPrisoner(final String prisonerIdentifier, final RequestToTransferOut requestToTransferOut) {
+        // check that prisoner is active in
+        final OffenderBooking booking = getAndCheckOffenderBooking(prisonerIdentifier);
+
+        final AgencyLocation toLocation = checkMovementTypes(TRN.getCode(), requestToTransferOut.getTransferReasonCode(), AgencyLocation.TRN);
+
+        // Generate the external movement out
+        final var movementReason = movementReasonRepository.findById(MovementReason.pk(requestToTransferOut.getTransferReasonCode())).orElseThrow(EntityNotFoundException.withMessage("No movement reason %s found", requestToTransferOut.getTransferReasonCode()));
+
+        final var transferDateTime = getAndCheckMovementTime(requestToTransferOut.getMovementTime(), booking.getBookingId());
+        // set previous active movements to false
+        setPreviousMovementsToInactive(booking);
+
+        createOutMovement(booking, TRN, movementReason, toLocation, transferDateTime, requestToTransferOut.getCommentText(), requestToTransferOut.getEscortType());
+        updateBeds(booking, transferDateTime);
+
+        // update the booking record
+        booking.setInOutStatus(TRN.getCode());
+        booking.setActiveFlag("N");
+        booking.setBookingStatus("O");
+        booking.setLivingUnitMv(null);
+        booking.setAssignedLivingUnit(null);
+        booking.setLocation(toLocation);
+        booking.setCreateLocation(toLocation);
+        booking.setStatusReason(TRN.getCode() + "-" + requestToTransferOut.getTransferReasonCode());
+        booking.setCommStatus(null);
+    }
+
+    private LocalDateTime getAndCheckMovementTime(final LocalDateTime movementTime, final Long bookingId) {
+        final var now = LocalDateTime.now();
```

Comment:
So from memory there is Clock bean in the API.  So something like this `LocalDate.now(clock)` makes it testable as well.